apollo-router/src/configuration/mod.rs

```diff
@@ -338,8 +338,7 @@ impl Graphql {
         }
     }
 
-    // Used in tests
-    #[allow(dead_code)]
+    #[cfg_attr(not(test), allow(dead_code))]
```

Comment:
Not quite what I meant....

I meant, let's just only compile the code for test. Thus
```
    #[cfg(test)]
    #[builder]
    pub(crate) fn fake_new(
        listen: Option<ListenAddr>,
```
That would have the effect of taking it out of the public API, which I think would be nice.

Reply:
Oh I see! The confusions stems from the fact I m not quite sure what parts we'll expose to plugin writers eventually and not, but that makes sense, fixing...

Reply:
Right. If we do decide to expose to plugin writers, we'll remove the #[cfg(test)] at the same time as the (crate) from the function definition.